tests/vm.rs

```diff
@@ -26,7 +26,7 @@ pub fn nested_attach_guard_should_not_detach_thread() {
         assert_eq!(val, 2);
```

Comment:
I'd check that the number of attached threads remains the same as before the second call (1?).